CONTRIBUTING.md

```diff
@@ -0,0 +1,330 @@
+    allowed (if they are of broader interest), but should be `#ifdef`-ed and not
+    change the code by default.
+
+- Code styling
+  - Consistent code styling is enforced with `make style` in the top-level
+    directory. This requires [Artistic Style](http://astyle.sourceforge.net) (we
+    specifically use version 2.05.1). See also the file `config/mfem.astylerc`.
+
+
+### Pull Requests
+
+- When your branch is ready for other developer to start looking / commenting on
+  the code, create a pull request towards `mfem:master`.
+
+- Pull request titles typically have the form:
+
+  `Description [new-feature-dev]`
+
+  for example:
+
+  `Parallel Unstructured Mesh Infrastructure (PUMI) integration [pumi-dev]`
+
+  Note the branch name suffix (in square brackets).
+
+- The title can contain a prefix in square brackets to emphasize the type of PR.
+  Some common choices are: `[DON'T MERGE]`, `[WIP]` and `[DISCUSS]`, for example:
+
+  `[DISCUSS] Hybridized DG [hdg-dev]`
+
+- Add a description, appropriate labels and assign yourself to the PR. The MFEM
+  team will add reviewers as appropriate.
+
+- List outstanding TODO items in the description, see PR #222 for an example.
+
+- Track the Travis CI and Appveyor [continuous integration](#automated-testing)
+  builds at the end of the PR. These should run clean, so try to address any
+  errors as soon as possible.
+
+
+## Pull Request Checklist
+
+Before a PR can be merged, it should satisfy the following:
+
+- [ ] Code builds
+- [ ] Code passes `make style`
+- [ ] Update the `CHANGELOG` file:
+    - [ ] Is this a new feature that the users need to be aware of? New or updated example or miniapp?
+    - [ ] Does it make sense to create a new section in the CHANGELOG to group this with other related features?
+- [ ] Update the `INSTALL` file :
+    - [ ] Has a new optional library been added?
+       - [ ] Make sure the external library is licensed under LGPL, not GPL!
+    - [ ] Does `make` or `cmake` have a new target?
+    - [ ] Did the requirements or the installation process change? (rare)
+- [ ] Update README:
+    - [ ] Does GLVis have new keys/functionality that needs to be documented?
+    - [ ] Is this a really major feature that needs mentioning in the short summary here? (rare)
+- [ ] New examples:
+    - [ ] All sample runs at the top of the example work
+    - [ ] Update examples/makefile
+      - [ ] Add the example code to the appropriate SEQ_EXAMPLES and PAR_EXAMPLES variables.
+      - [ ] Add any files generated by it to the clean target.
+    - [ ] Update examples/CMakeLists.txt
+      - [ ] Add the example code to the ALL_EXE_SRCS variable.
+      - [ ] Make sure THIS_TEST_OPTIONS is set correctly for the new example.
+- [ ] List the new example in doc/CodeDocumentation.dox
+- [ ] Companion pull request for documentation in mfem/web repo
+   - [ ] Update or add miniapp-specific documentation, see e.g. the src/meshing.md and src/electromagnetics.md.
```

Comment:
Should this read "Update or add example-specific..."?

Reply:
Thanks @mlstowell! 

Feel free to directly edit (and expand) as you see fit :)